src/util/schema_keywords.js

```diff
@@ -0,0 +1,174 @@
+                    Number.isInteger(${d}) &&
+                    !isNaN(new Date(${d}).getTime())
+                `) // good idate
+                .elseIf(CG` ${d} instanceof Date `) // ok but see todo ^
+                .else()
+                .code(() => cxt.error())
+                .endIf();
+        }
+    },
+
+    // schema: { objectid: true } will match (new mongodb.ObjectId()) or (new mongodb.ObjectId()).valueOf()
+    objectid: {
+        keyword: 'objectid',
+        // schemaType: 'boolean',
+        /**
+         * 
+         * @param {KeywordCxt} cxt 
+         * 
+         */
+        code(cxt) {
+            const d = cxt.it.data;
+            cxt.gen
+                .if(CG`
+                    typeof ${d} === 'object' &&
+                    ${d} && 
+                    ${d}.constructor && 
+                    ${d}.constructor.name === 'ObjectID'
+                `)
+                .elseIf(CG`
+                    typeof ${d} === 'string' &&
+                    /^[0-9a-fA-F]{24}$/.test(${d})
+                `)
+                .else()
+                .code(() => cxt.error())
+                .endIf();
+        }
+    },
+
+    // schema: { binary: 64 } will match Buffer.alloc(64)
+    // schema: { binary: true } will match Buffer.alloc(<any>)
+    binary: {
+        keyword: 'binary',
+        // schemaType: ['boolean', 'number'],
+        /**
+         * 
+         * @param {KeywordCxt} cxt 
+         * 
+         */
+        code(cxt) {
+            const d = cxt.data;
+            const s = cxt.schemaValue;
+            cxt.gen
+                .if(CG`
+                    Buffer.isBuffer(${d}) && (
+                        typeof ${s} === 'number' ? 
+                        ${d}.length === ${s} :
+                        ${s}
+                    )
+                `)
+                .elseIf(CG`
+                    ${d}._bsontype === 'Binary' && (
+                        typeof ${s} === 'number' ? 
+                        ${d}.length() === ${s} :
+                        ${s}
+                    )
+                `)
+                .else()
+                .code(() => cxt.error())
+                .endIf();
+        }
+    },
+
+    wrapper: {
+        keyword: 'wrapper',
+        errors: false,
+        modifying: true,
+        /**
+         * 
+         * @param {KeywordCxt} cxt 
+         * 
+         */
+        code(cxt) {
+            if (cxt.it.dataLevel <= 0) return;
+            const d = cxt.data;
+            const s = cxt.schemaValue;
+            const p = cxt.it.parentData;
+            const pp = cxt.it.parentDataProperty;
+            cxt.gen
+                .if(CG` ${d} instanceof ${s} `)
+                .elseIf(CG` ${s}.can_wrap(${d}) `)
+                .assign(CG` ${p}[${pp}] `, CG` new ${s}(${d}) `)
+                .else()
+                .code(() => cxt.error())
+                .endIf();
+        }
+    },
+
+    wrapper_check_only: {
+        keyword: 'wrapper_check_only',
```

Comment:
I think and we should set the keyword name here to `wrapper` too and not wrapper_check_only because it is used as a drop-in replacement for the original wrapper keyword, and no api/schema actually uses `wrapper_check_only` to define a property
```suggestion
        keyword: 'wrapper',
```